### PR TITLE
Add galasactl secrets get command and formatters

### PR DESCRIPTION
**Pull request by eamansour**
## Why?
See https://github.com/galasa-dev/projectmanagement/issues/1467

## Changes
- Added `galasactl secrets get` command to retrieve all secrets from a Galasa service
  - Includes a `--name` flag to get an individual secret, and `--format` to display results in either `summary` or `yaml` formats

**This PR builds upon changes in https://github.com/galasa-dev/cli/pull/300 - that PR should be delivered first, then this PR can be rebased to remove the duplicate changes**

**Builds for this PR will fail until https://github.com/galasa-dev/framework/pull/662 is delivered**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-301-6gjh5 for details. If you are unable to do so, please contact a member of the Galasa team.

File: pkg/secrets/secrets.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "strings"
+
+    galasaErrors "github.com/galasa-dev/cli/pkg/errors"
+)
+
+func validateSecretName(secretName string) error {
+    var err error
+    secretName = strings.TrimSpace(secretName)
+
+    if secretName == "" || strings.ContainsAny(secretName, " \n\t") {
```
Comment (techcobweb) [inline pkg/secrets/secrets.go:19]: I should think we need to validate it more than that ? 
eg: "a..secret....with.lots.of.dots" will blow up etcd no ?
Latin-1 characters only ? what about - and _ are they allowed ?

File: pkg/secretsformatter/yamlFormatter.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+		var yamlRepresentationBytes []byte
+		yamlRepresentationBytes, err = yaml.Marshal(secret)
+		if err == nil {
+			yamlStr := string(yamlRepresentationBytes)
+
+			// The generated bean serialises in json as 'apiVersion' which is correct. In yaml it serialises as 'apiversion' (incorrect)
+			// So this is a hack to correct that failure.
+			// Note: This will corrupt any value string which also has 'apiversion' inside it !
+			// TODO: The fix is to change the bean and add a 'yaml' annotation so it gets rendered correctly. Golang has yaml annotations, but does the generator support them ?
```
Comment (techcobweb) [inline pkg/secretsformatter/yamlFormatter.go:49]: TODO left in the code ?

**Review by techcobweb — APPROVED** [review]
Minor comments but approved

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-cli-pr-301-z5kl6 for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Pull request merged**